src/Core/Service/ServiceRegistry/SaveConsentRequest.php

```diff
@@ -0,0 +1,36 @@
+<?php declare(strict_types=1);
+
+namespace Shopware\Core\Service\ServiceRegistry;
+
+use Shopware\Core\Framework\Log\Package;
+
+/**
+ * @internal
+ *
+ * @codeCoverageIgnore
+ */
+#[Package('framework')]
+class SaveConsentRequest implements \JsonSerializable
+{
+    public function __construct(
+        public string $identifier,
+        public string $consentingUserId,
+        public string $shopIdentifier,
+        public string $consentDate,
+        public string $consentRevision,
+        public ?string $licenseHost = null,
+    ) {
+    }
+
+    public function jsonSerialize(): mixed
+    {
+        return [
+            'identifier' => $this->identifier,
+            'consentingUserId' => $this->consentingUserId,
+            'shopIdentifier' => $this->shopIdentifier,
+            'consentDate' => $this->consentDate,
+            'consentRevision' => $this->consentRevision,
+            'licenseHost' => $this->licenseHost,
+        ];
```

Comment:
```suggestion
        return get_object_vars($this);
```

or also the JsonSerializableTrait

Reply:
Same https://github.com/shopware/shopware/pull/10672#discussion_r2180022293 - I think it's better if there's less dependency.